tools/autograd/gen_inplace_or_view_type.py

```diff
@@ -28,10 +28,19 @@
 #
 # A map: function name => name of the argument that all outputs are view of
 
-VIEW_FUNCTIONS_WITH_METADATA_CHANGE = ['view_as_complex', 'view_as_real', '_conj', '_neg_view']
+VIEW_FUNCTIONS_WITH_METADATA_CHANGE = [
+    'view_as_complex',
+    'view_as_real',
+    '_conj',
+    '_neg_view',
+    'matrix_H',
+    'mH',
+    'adjoint'
+]
 
 VIEW_FUNCTIONS = {
     'numpy_T': 'self',
+    'mT': 'self',
```

Comment:
again same not sure if you need this since it's a composite op cc. @albanD

Reply:
synced offline -- @Lezcano since all these functions are composite ops they don't need to be added in `VIEW_FUNCTIONS` or `RETURNS_VIEWS_OF_INPUT `, instead we only need entries in `RETURNS_VIEWS_OF_INPUT` below